### ISSUE TITLE
browser.js does not work in strict mode.

### ISSUE THREAD

**Issue by iliakan** [websockets/ws#478]
It tries to get `global` from a `(function() { ... this }())`, but in `use strict` such functions execute with `this = undefined`, not `window`.

**Comment by 3rd-Eden** [comment]
But the browser file doesn't include any strict mode so it works. Unless you some how forced strict module globally on all files. Which is recommended by nobody. So as far as I'm concerned, this is a non issue.

**Comment by iliakan** [comment]
This suddenly became an issue to me, because ES6-modules have "implicit strict mode".

I'm using babel.js for ES6, and it built my project (including `ws` browser.js) with all-around-strict-modes. So browser.js got "use strict" at the top too.

After debugging I found out the reason and excluded browser.js from babel.js compilation (not needed for sure), so the issue was fixed.

But in the view of ES6 modules it might be future-proof to make the module strict-mode-compatible.

**Comment by 3rd-Eden** [comment]
Could you confirm that the `browserefactor` branch works as expected and correctly resolved your bug?

**Comment by 3rd-Eden** [comment]
But in any case, your problem is also more a babel bug then a ws bug. As ES6 strict mode should only be applied to actual ES6 modules and ES6 classes. This file follows non of those patterns and should therefor be excluded from the strict-mode.

**Comment by iliakan** [comment]
I totally agree with you.

I fixed the issue by excluding the file from babel.

After consideration I thought that future-wise it may be better to make the file strict-mode compatible and embeddable in ES6 modules. And posted the issue.

My refactorings led me far away, the code which caused the problem is replaced, but I guess it should work.

**Comment by noraesae** [comment]
+1 for the fix too.

I understand the function-level strict-mode is recommended. However, from Node.js v4.0, some ES6 features, such as block-scoped declarations, can only be used in the strict mode. It means that if I'd like to declare a top-level class like below, I need to add `'use strict';` on the top.

```js
const ws = require('ws');
class A { } // doesn't work
```

```js
'use strict';
const ws = require('ws');
class A { } // works
```

I hope you understand that it's really annoying to create a functional scope every time we declare a class, especially writing a module in Node.js. And if somebody want to use the same module in a browser too, which is extremely common now with Browserify or Webpack, there comes the problem.

In summary, I love ws' simple API but not working in the strict mode really stop me from using it. Please consider fixing it to work with it.

Many thanks!
